bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -277,71 +320,240 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
+      ModifyColumnFamiliesRequest.Builder modifyColumnBuilder = ModifyColumnFamiliesRequest
+          .newBuilder().addModifications(modification).setName(toBigtableName(tableName));
+      return modifyColumnBuilder;
+    }).thenCompose(
+        d -> FutureUtils.toCompletableFuture(
+            bigtableTableAdminClient.modifyColumnFamilyAsync(d.build()))
+            .thenApply(r -> null));
   }
 
   @Override
-  public CompletableFuture<Void> deleteSnapshot(String arg0) {
-    throw new UnsupportedOperationException("deleteSnapshot"); // TODO
-
+  public CompletableFuture<Void> deleteSnapshot(String snapshotName) {
+    return CompletableFuture.supplyAsync(() -> {
+      try {
+        return DeleteSnapshotRequest.newBuilder()
+          .setName(getClusterName().toSnapshotName(snapshotName))
+          .build();
+      } catch (IOException e) {
+        throw new CompletionException(e); 
+      }
+    }).thenCompose(
+        d -> FutureUtils.toCompletableFuture(
+            bigtableTableAdminClient.deleteSnapshotAsync(d)).thenApply(r -> null));
   }
 
   @Override
-  public CompletableFuture<Void> deleteTableSnapshots(Pattern arg0, Pattern arg1) {
-    throw new UnsupportedOperationException("deleteTableSnapshots"); // TODO
+  public CompletableFuture<Void> deleteTableSnapshots(Pattern tableNamePattern,
+      Pattern snapshotNamePattern) {
+    return listTableSnapshots(tableNamePattern, snapshotNamePattern).thenApply(r->{
+      for (SnapshotDescription snapshotDescription : r) {
+        deleteSnapshot(snapshotDescription.getName());
+      }
+      return null;
+    });
   }
 
   @Override
-  public CompletableFuture<Void> modifyColumnFamily(TableName arg0, ColumnFamilyDescriptor arg1) {
-    throw new UnsupportedOperationException("modifyColumnFamily"); // TODO
+  public CompletableFuture<Void> modifyColumnFamily(TableName tableName, ColumnFamilyDescriptor columnFamilyDesc) {
+    return CompletableFuture.supplyAsync(() -> {
+      return Modification.newBuilder()
+          .setId(columnFamilyDesc.getNameAsString())
+          .setCreate(new ColumnDescriptorAdapter().adapt((HColumnDescriptor) columnFamilyDesc)
+          .build());
+    }).thenCompose(
+        d -> modifyColumn(tableName, columnFamilyDesc.getNameAsString(), "update", d)).
+        thenApply(r -> null);
   }
 
   @Override
-  public CompletableFuture<Void> restoreSnapshot(String arg0) {
-    throw new UnsupportedOperationException("restoreSnapshot"); // TODO
+  public CompletableFuture<Void> restoreSnapshot(String snapshotName) {
+  	boolean takeFailSafeSnapshot = BigtableConstants.DEFAULT_SNAPSHOT_RESTORE_TAKE_FAILSAFE_SNAPSHOT;
+    return restoreSnapshot(snapshotName, takeFailSafeSnapshot);
   }
 
   @Override
-  public CompletableFuture<Void> restoreSnapshot(String arg0, boolean arg1) {
-    throw new UnsupportedOperationException("restoreSnapshot"); // TODO
+  public CompletableFuture<Void> restoreSnapshot(String snapshotName, boolean takeFailSafeSnapshot) {
```

Comment:
Can you please add comments here?  Also, it may be worth breaking up this method into smaller methods.  There seems to be a need for a method like `snapshotExists()`